src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -2180,4 +2188,135 @@ inline void rpp_resize_store_pkd3(Rpp8s *dstPtr, __m128 *p)
     rpp_store48_f32pln3_to_i8pkd3(dstPtr, p);
 }
 
+inline RppStatus rpp_nn_load_u8pkd3(Rpp8u *srcRowPtrsForInterp, Rpp32s *loc, __m128i &p)
```

Comment:
@fiona-gladwin In the previous u8 to f32 bilinear loads, you were using xmm_pxConvertI8 to do the I8 offset. NN interp doesn't need that, so many of the u8 and i8 helpers end up being the same except for the datatype itself which is Rpp8u/8s. Have you tried to template them to avoid multiple instances?

Reply:
Yes, but this function is being called using the rpp_simd_load, hence I was unable to pass to a templated inline function